pomcorn/component.py

```diff
@@ -177,21 +177,37 @@ class ListComponent(Generic[ListItemType, TPage], Component[TPage]):
     * all
     * get_item_by_text()
 
-    Waits for `base_item_locator` property to be implemented and value of
-    `item_class` to be set.
-
     Waits for `base_item_locator` property  to be overridden or one of the
     attributes (`item_locator` or `relative_item_locator`) to be specified.
 
-    The `item_class` attribute is also required.
-
     """
 
-    item_class: type[ListItemType]
-
     item_locator: locators.XPathLocator | None = None
     relative_item_locator: locators.XPathLocator | None = None
 
+    def __init__(
+        self,
+        page: TPage,
+        base_locator: locators.XPathLocator | None = None,
+        wait_until_visible: bool = True,
+    ):
+        super().__init__(page, base_locator, wait_until_visible)
+        if item_class := getattr(self, "item_class", None):
+            import warnings
+
+            warnings.warn(
+                DeprecationWarning(
+                    "\nSpecifying `item_class` attribute in `ListComponent` "
+                    f"({self.__class__}) is DEPRECATED. It is now "
+                    "automatically substituted from Generic[ListItemType]. "
+                    "Ability to specify this attribute will be removed soon.",
+                ),
+                stacklevel=2,
+            )
+            self._item_class = item_class
+        else:
+            self._item_class = self._get_list_item_class()
```

Comment:
I think since pomcorn is pre-release, maybe it's okay to just drop it

Reply:
Then we would have to update all of our current projects... It's not difficult, but it will takes time